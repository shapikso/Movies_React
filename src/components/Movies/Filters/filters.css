@keyframes active {
    from { transform: scale(0); }
    to { transform: scale(1); }
}

.form-filters {
    position: absolute;
    top: 100px;
    padding: 40px;
    -webkit-backdrop-filter: blur(50px);
    backdrop-filter: blur(50px);
    min-width: 560px;
    max-width: 1020px;
    width: 100%;
    display: flex;
    flex-direction: column;
    justify-content: flex-start;

    max-height: 90%;
    overflow-y: auto;
}

.form-filters::-webkit-scrollbar {
    display: none;
}

.form-filters__header {
    display: flex;
    justify-content: space-between;
    flex-wrap: wrap;
}

.form-filters__header >:not(:last-child) {
    margin-right: 25px;
}

.form-filters__header * {
    margin-bottom: 40px;
}

.form-filters select {
    -webkit-appearance: none;
    -moz-appearance: none;
    appearance: none;
    /*background: url("../../../filters/arrow-down.svg") no-repeat 15px center;*/
    padding-left: 40px;
}

.form-filters select:hover {
    cursor: pointer;
}

.form-filters select option {
    color: var(--dark-primary-color);
}

.form-filters__ranges {
    display: flex;
    justify-content: space-between;
    flex-wrap: wrap;
}

.form-filters__ranges >:not(:last-child) {
    margin-right: 20px;
}

.form-filters__ranges__item {
    margin-bottom: 40px;
}

.double-range {
    border: 1px solid var(--light-primary-color);
    border-radius: 5px;
    padding: 14px;
}

.double-range >:not(:last-child) {
    margin-bottom: 15px;
}

.double-range__info {
    width: 250px;
    display: flex;
    justify-content: space-between;
    align-items: center;
}

.double-range__info-min,
.double-range__info-max {
    width: 97%;
    background: none;
    color: var(--light-primary-color);
    padding: 10px;
    text-align: center;
}

.double-range__info-max {
    text-align: right;
}

.double-range__place {
    width: 100%;
    position: relative;
    height: 20px;
}

.double-range__place:before {
    content: "";
    width: 100%;
    height: 2px;
    position: absolute;
    top: 50%;
    transform: translateY(-50%);
    left: 0;
    background-color: var(--light-primary-color);
}

.double-range__place-item {
    position: absolute;
    left: 0;
    top: 50%;
    transform: translateY(-50%);
    -webkit-appearance: none;
    -moz-appearance: none;
    appearance: none;
    width: 100%;
    height: 0;
    padding: 0;
    margin: 0;
    border: none;
    outline: none;
    background: none;
}

.double-range__place-item::-webkit-slider-runnable-track {
    width: 100%;
    height: 0;
    cursor: pointer;
    background-color: var(--light-primary-color);
    box-shadow: none;
    border: none;
    outline: none;
    z-index: 0;
}

.double-range__place-item::-webkit-slider-thumb {
    z-index: 2;
    position: relative;
    height: 18px;
    width: 18px;
    border-radius: 50%;
    background-color: var(--light-primary-color);
    cursor: pointer;
    -webkit-appearance: none;
    appearance: none;
    margin-top: -8px;
}

.double-range__place-item::-moz-range-track {
    width: 100%;
    height: 0;
    cursor: pointer;
    background-color: var(--light-primary-color);
    box-shadow: none;
    border: none;
    outline: 0;
    margin: 0;
    padding: 0;
    z-index: 0;
}

.double-range__place-item::-moz-range-thumb {
    z-index: 2;
    position: relative;
    height: 18px;
    width: 18px;
    border-radius: 50%;
    border: none;
    background-color: var(--light-primary-color);
    cursor: pointer;
}

.form-filters__release-date {
    display: flex;
    justify-content: space-between;
    margin-bottom: 40px;
    width: 880px;
}

.form-filters__release-date__item {
    width: 48%;
}

.form-filters__buttons {
    display: flex;
    justify-content: center;
    position: relative;
    gap: 40px;
}

.form-filters__buttons button[type="reset"] {
    min-width: 0;
    max-width: 100px;
    width: 100%;
    position: absolute;
    top: 0;
    right: -285px;
}

.date-picker {
    position: relative;
}

.date-picker__input {
    width: 100%;
}

.date-picker__input[type="text"]:focus {
    box-shadow: none;
}

.date-picker__clear {
    width: 50px;
    height: 50px;
    position: absolute;
    top: 0;
    right: 0;

    display: flex;
    justify-content: center;
    align-items: center;
    border-left: 1px solid var(--light-primary-color);
    border-radius: 0 5px 5px 0;
    transition: all 0.2s ease;
}

.date-picker__clear i {
    width: 100%;
    height: 100%;
    display: flex;
    justify-content: center;
    align-items: center;
}

.date-picker__clear i svg {
    display: block;
    transition: all 0.2s ease;
}

.date-picker__clear:hover i svg {
    fill: var(--primary-color);
}

.date-picker__clear:active i svg {
    fill: var(--light-primary-color);
}

.date-picker__clear:hover {
    cursor: pointer;
    background-color: var(--light-primary-color);
}

.date-picker__clear:active {
    background-color: var(--primary-color-opacity);
}

.frosted-glass-effect {
    background: linear-gradient(
            155deg
            , rgba(255, 255, 255, 0.2) -50%, rgba(255, 255, 255, 0.05) 80%);
    border: 1px solid rgba(255, 255, 255, 0.3);
    border-radius: 25px;
    -webkit-backdrop-filter: blur(10px);
    backdrop-filter: blur(10px);
}

.basic-field {
    padding: 10px 25px;
    background: none;
    border: 1px solid var(--light-primary-color);
    border-radius: 5px;
    height: 50px;
    font-size: 16px;
    color: var(--light-primary-color);
}
.sliderArea {
    width: 300px;
    margin: 0 auto;
}
.close {
    cursor: pointer;
    font-size: 30px;
    min-width: 20px;
    position: absolute;
    right: -340px;
    top: -480px;
}

