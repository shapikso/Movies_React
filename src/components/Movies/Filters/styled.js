import styled from "styled-components";

export const StFormFilter = styled.form`
  position: absolute;
  top: 100px;
  padding: 40px;
  -webkit-backdrop-filter: blur(50px);
  backdrop-filter: blur(50px);
  min-width: 560px;
  max-width: 1020px;
  width: 100%;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  max-height: 90%;
  overflow-y: auto;
  background: linear-gradient(
          155deg
          , rgba(255, 255, 255, 0.2) -50%, rgba(255, 255, 255, 0.05) 80%);
  border: 1px solid rgba(255, 255, 255, 0.3);
  border-radius: 25px;
  -webkit-backdrop-filter: blur(10px);
  backdrop-filter: blur(10px);
  `;

export const StClose = styled.span`
  cursor: pointer;
  font-size: 30px;
  min-width: 20px;
  position: absolute;
  right: 5px;
  top: 5px;
    `;

export const StFilterRow = styled.div`
  display: flex;
  justify-content: space-between;
  flex-wrap: wrap;
  margin-bottom: 40px;
  gap: 25px;
  align-items: center;
    `;
export const StFilterButton = styled.div`
  display: flex;
  justify-content: center;
  position: relative;
  gap: 40px;
  button[type="reset"] {
    min-width: 0;
    max-width: 100px;
    width: 100%;
  }
    `;

export const StInputWrapper = styled.div`
    width: 500px;
 `;