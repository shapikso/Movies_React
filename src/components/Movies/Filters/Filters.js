import React, {Component} from 'react';
import './filters.css';
import Select from "./Selector/Select";
import {movieLanguages,movieStatus} from '../../../constants/selectors';
import FilterInput from "./Input/FilterInput";
import DoubleRange from "./DoubleRange/DoubleRange";
import DatePicker from "react-datepicker";
import "react-datepicker/dist/react-datepicker.css";
import Button from "../../common/Button/Button";

class Filters extends Component {
    constructor(props) {
        super(props);
        this.state = {
            searchTitle: '',
            language: '',
            status:  '',
            minDate: '',
            maxDate: '',
            minBudget: 0,
            maxBudget: 500000000,
            minPopularity: 0,
            maxPopularity: 200,
            minVote: 0,
            maxVote: 30000
        };
    }
    setBudget = (minBudget, maxBudget) => this.setState({minBudget: minBudget,maxBudget: maxBudget});
    setPopularity = (minPopularity, maxPopularity) => this.setState({minPopularity: minPopularity,maxPopularity: maxPopularity});
    setVote = (minVote, maxVote) => this.setState({minVote: minVote,maxVote: maxVote});
    setLanguage = (language) => this.setState({language: language});
    setStatus = (status) => this.setState({status: status});
    setSearchTitle = (title) => this.setState({searchTitle: title});
    setMinDate = (date) => this.setState({minDate: date});
    setMaxDate = (date) => this.setState({maxDate: date});

    render() {
        return (
            <form id="filters" name="filters" className="form-filters frosted-glass-effect" action="#">
                <div className="form-filters__header">
                    <FilterInput setSearchTitle={this.setSearchTitle} />
                    <Select setDelector={this.setStatus} options={movieStatus} />
                    <Select setDelector={this.setLanguage} options={movieLanguages} />
                </div>
                <div className="form-filters__ranges">
                    <DoubleRange setRange={this.setBudget}
                        title="Budget, $"
                        step="100000"
                        minValue="0"
                        maxValue="500000000"
                        maxInputValue={this.state.maxBudget}
                        minInputValue={this.state.minBudget}
                    />
                    <DoubleRange setRange={this.setPopularity}
                        title="Popularity"
                        step="1" minValue="0"
                        maxValue="200"
                        maxInputValue={this.state.maxPopularity}
                        minInputValue={this.state.minPopularity}
                    />
                    <DoubleRange setRange={this.setVote}
                        title="Vote count"
                        step="1"
                        minValue="0"
                        maxValue="30000"
                        maxInputValue={this.state.maxVote}
                        minInputValue={this.state.minVote}
                    />
                </div>
                <div className="form-filters__release-date">
                    <div id="release-date-first" className="form-filters__release-date__item date-picker">
                        <DatePicker selected={this.state.minDate}
                            onChange={(date) => this.setMinDate(date)}
                            placeholderText="Select Date.."
                            className="basic-field date-picker__input" />
                    </div>
                    <div id="release-date-last" className="form-filters__release-date__item date-picker">
                        <DatePicker selected={this.state.maxDate}
                            onChange={(date) => this.setMaxDate(date)}
                            placeholderText="Select Date.."
                            className="basic-field date-picker__input" />
                    </div>
                </div>
                <div className="form-filters__buttons">
                    <Button type="submit" className="basic-btn" contentKey ="Submit"/>
                    <Button type="reset" className="basic-btn" isDisabled={true} contentKey="Reset"/>
                </div>
            </form>
        );
    }
}

export default Filters;