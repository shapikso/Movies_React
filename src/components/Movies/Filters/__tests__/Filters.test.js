import React from 'react';
import {shallow, mount} from 'enzyme';
import Filters from "../Filters";

describe('Filters', ()=> {
    const props = {
        setFilter: jest.fn(),
        closeModal: jest.fn(),
        onSubmite: jest.fn(),
        clearFilters: jest.fn(),
        filters: {
            title: 'aaaaa',
            language: 'en',
            status: 'Released',
            release_date_first: '',
            release_date_last: '',
            budget_min: 0,
            budget_max: 5000000,
            popularity_min: 0,
            popularity_max: 5,
            revenue_min: 0,
            revenue_max: 300,
        }
    };
    it('should render correctly', () => {
        const component = shallow(<Filters {...props} />);
        expect(component).toMatchSnapshot();
    });
    it('should clear filters', () => {
        const component = mount(<Filters {...props} />);
        component.find('button').at(0).getElement().props.onClick();
        expect(props.clearFilters).toHaveBeenCalled();
    });
    it('should check on submit button', () => {
        const component = mount(<Filters {...props} />);
        component.find('button').at(1).getElement().props.onClick();
        expect(props.closeModal).toHaveBeenCalled();
        expect(props.onSubmite).toHaveBeenCalled();
    });
    it('should render prop setFilter correctly', () => {
        const component = shallow(<Filters {...props} />);
        expect(component.find('FilterHeaders').getElement().props.setFilter).toEqual(props.setFilter);
    });
});