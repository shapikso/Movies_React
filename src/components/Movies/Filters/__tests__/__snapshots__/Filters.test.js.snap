// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters should render correctly 1`] = `ShallowWrapper {}`;
